state/src/account.rs

```diff
@@ -0,0 +1,395 @@
+        assert_eq!(a.code_size, 1);
+        a.commit_code(&mut db).unwrap();
+        assert_eq!(
+            a.code_hash,
+            "78fe1396dda648dcbccc3c17af4cd29de873f2cdf5e4c5eb04e0ef08e86cc267".into()
+        );
+        assert_eq!(db.get(&a.code_hash()).unwrap().unwrap(), vec![0x55]);
+    }
+
+    #[test]
+    fn state_object_storage_1() {
+        let mut a = StateObject::new(69u8.into(), 0.into());
+        let mut db = cita_trie::db::MemoryDB::new();
+        a.set_storage(0.into(), 0x1234.into());
+        a.commit_storage(&mut db).unwrap();
+        assert_eq!(
+            a.storage_root,
+            "ca8f89e4444c7453e96568511298af8049553232cfdb9255be8799d68c28b297".into()
+        );
+    }
+
+    #[test]
+    fn state_object_storage_2() {
+        let mut a = StateObject::new(69u8.into(), 0.into());
+        let mut db = cita_trie::db::MemoryDB::new();
+        a.set_storage(0.into(), 0x1234.into());
+        a.commit_storage(&mut db).unwrap();
+        assert_eq!(
+            a.storage_root,
+            "ca8f89e4444c7453e96568511298af8049553232cfdb9255be8799d68c28b297".into()
+        );
+        a.set_storage(1.into(), 0x1234.into());
+        a.commit_storage(&mut db).unwrap();
+        assert_eq!(
+            a.storage_root,
+            "41cf81d2e6063cccd6965e9ca7d2b2ca95c6cf68012c9ac0be8564fd30e106b8".into()
+        );
+        a.set_storage(1.into(), 0.into());
+        a.commit_storage(&mut db).unwrap();
+        assert_eq!(
+            a.storage_root,
+            "ca8f89e4444c7453e96568511298af8049553232cfdb9255be8799d68c28b297".into()
+        );
+    }
+
+    #[test]
+    fn state_object_storage_3() {
+        let mut a = StateObject::new(69u8.into(), 0.into());
+        let mut db = cita_trie::db::MemoryDB::new();
+        let a_rlp = {
+            a.set_storage(0x00u64.into(), 0x1234u64.into());
+            a.commit_storage(&mut db).unwrap();
+            a.init_code(vec![]);
+            a.commit_code(&mut db).unwrap();
+            rlp::encode(&a.account())
+        };
+        a = StateObject::from_rlp(&a_rlp[..]).unwrap();
+        assert_eq!(
+            a.storage_root,
+            "ca8f89e4444c7453e96568511298af8049553232cfdb9255be8799d68c28b297".into()
+        );
+        assert_eq!(
+            a.get_storage(&mut db, &0x00u64.into()).unwrap().unwrap(),
+            0x1234u64.into()
+        );
+        assert_eq!(a.get_storage(&mut db, &0x01u64.into()).unwrap(), None);
+    }
+
+    #[test]
+    fn state_object_note_code() {
+        let mut a = StateObject::new(69u8.into(), 0.into());
+        let mut db = cita_trie::db::MemoryDB::new();
+        let a_rlp = {
+            a.init_code(vec![0x55, 0x44, 0xffu8]);
+            a.commit_code(&mut db).unwrap();
+            a.rlp()
+        };
+        a = StateObject::from_rlp(&a_rlp[..]).unwrap();
+        a.read_code(&mut db).unwrap();
+        assert_eq!(a.code, vec![0x55, 0x44, 0xffu8]);
+    }
+}
```

Comment:
There are two kinds of hard code in the tests module: code_hash and storage_root. I think the first one can be avoided, I am not sure how to deal with the second one.